### DISCUSSION TITLE
Improve Go License/DependsOn scanning

### DISCUSSION THREAD

**Discussion by no10xcoder** [aquasecurity/trivy#7354] [Ideas]
### Description

Currently the Go scanner only detects licenses and transitive dependencies when the Go modules have been downloaded (see [here](https://github.com/aquasecurity/trivy/blob/0c6687d5ba0df5de2632aad26c21ea99565bd9e1/pkg/fanal/analyzer/language/golang/mod/mod.go#L131) and [here](https://github.com/aquasecurity/trivy/blob/0c6687d5ba0df5de2632aad26c21ea99565bd9e1/pkg/fanal/analyzer/language/golang/mod/mod.go#L186))

For the license scanning I think it's also possible to scan the `vendor` folder when it's available. Let me know if you want me to draft an MR for that. 

For the transitive dependencies it's also a possibility to run `go mod download`, but I would understand if that goes a bit too far.

### Target

SBOM

### Scanner

License

**Comment by knqyf263** [comment]
>For the license scanning I think it's also possible to scan the vendor folder when it's available. Let me know if you want me to draft an MR for that.

Yes, please. Also, want to know how many users are still using `vendor` dir.

>For the transitive dependencies it's also a possibility to run go mod download, but I would understand if that goes a bit too far.

Trivy doesn't run an external command. Please let us know if you have any ideas.
https://aquasecurity.github.io/trivy/v0.54/community/principles/#external-dependency-free-single-binary
